sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2261,6 +2261,24 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     assert(df.queryExecution.executedPlan.isInstanceOf[WholeStageCodegenExec])
   }
 
+  test("SPARK-24341: IN subqueries with struct fields") {
```

Comment:
I cannot really add them there since I need to intercept `AnalysisException` here, but if you have suggestions about better places for this, I am happy to move it.